### PR TITLE
input-field: fix missing `fail_text` when `placeholder_text` is empty

### LINKED ISSUE #178: [BUG] fail_text not drawn if placeholder_text is empty
It seems that when `input-field:placeholder_text` is left empty,  the `input-field:fail_text` doesn't show up.

When I set `placeholder_text = x`, the fail text does show up properly on a failure. However when I just have `placeholder_text = ` in my config, the fail text isn't showing up.

Note: If anyone else is running into this and wants a quick fix, you can do:
```
  placeholder_text = ‍# There is a zero-width joiner character here (U+200D) since fail_text doesn't work with this empty
```

### PR DESCRIPTION
**Pull request by Sasikuttan2163**
Closes #178

### CODE REVIEW

**Review by vaxerski — APPROVED** [review]

File: src/renderer/widgets/PasswordInputField.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ CPasswordInputField::CPasswordInputField(const Vector2D& viewport_, const std::u
     g_pHyprlock->m_bNumLock = outerColor.invertNum;
 
     std::string placeholderText = std::any_cast<Hyprlang::STRING>(props.at("placeholder_text"));
-    if (!placeholderText.empty()) {
+    if (!placeholderText.empty() || !configFailText.empty()) {
```
[the remark this answers is not in the document]
Comment (Sasikuttan2163) [reply src/renderer/widgets/PasswordInputField.cpp:50]: The fail text replaces the placeholder text, so if the placeholder text doesn't render, fail text doesn't too.

[the remark this answers is not in the document]
Comment (Sasikuttan2163) [reply src/renderer/widgets/PasswordInputField.cpp:50]: [Here](https://github.com/hyprwm/hyprlock/blob/1c92c6109c2ef45085e0541c24f6324fd382ac5b/src/renderer/widgets/PasswordInputField.cpp#L336) is where it happens

[the remark this answers is not in the document]
Comment (vaxerski) [reply src/renderer/widgets/PasswordInputField.cpp:50]: hm, I guess it's an ok workaround. just add a comment for that, it's weird to check for that

[the remark this answers is not in the document]
Comment (Sasikuttan2163) [reply src/renderer/widgets/PasswordInputField.cpp:53]: Hope it's cool now

[the remark this answers is not in the document]
Comment (bvr-yr) [reply src/renderer/widgets/PasswordInputField.cpp:53]: yeah the whole block is skipped
https://github.com/hyprwm/hyprlock/blob/1c92c6109c2ef45085e0541c24f6324fd382ac5b/src/renderer/widgets/PasswordInputField.cpp#L280
and no `forceReload`
https://github.com/hyprwm/hyprlock/blob/1c92c6109c2ef45085e0541c24f6324fd382ac5b/src/renderer/widgets/PasswordInputField.cpp#L300-L301

**Review by vaxerski — APPROVED** [review]

**Pull request merged**